### PR TITLE
(feat) add chaosengine construction doc

### PR DESCRIPTION
**Pull request by ksatchit**
- Introduce a concepts section with a doc on construction of chaosengine resource

Signed-off-by: ksatchit <ksatchit@mayadata.io>

### CODE REVIEW

File: docs/chaosengine-concepts.md [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+  <td><The <code>jobCleanupPolicy</code> controls whether or not the experiment pods are removed once execution completes. Set to <code>retain</code> for debug purposes (in the absence of standard logging mechanisms).</td>
+</tr>
+</table>
+
+## Component Specification
+
+<table>
+<tr>
+  <th>Field</th>
+  <td><code>.spec.components.runner.image</code></td>
```
Comment (ispeakc0de) [inline docs/chaosengine-concepts.md:274]: Why are we not adding `.spec.components.monitor.image` field?

File: docs/chaosengine-concepts.md [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+<tr>
+  <th>Notes</th>
+  <td>The <code>appns</code> in the spec specifies the namespace of the AUT. Usually provided as a quoted string.</td>
+</tr>
+</table>
+
+<table>
+<tr>
+  <th>Field</th>
+  <td><code>.spec.appinfo.applabel</code></td>
```
Comment (ispeakc0de) [inline docs/chaosengine-concepts.md:77]: Why are we not adding `.spec.appinfo.appkind` field? As we are using appkind in the annotationCheck

**Review by ispeakc0de — CHANGES_REQUESTED** [review]

File: docs/chaosengine-concepts.md [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+  <td>Optional</td>
+</tr>
+<tr>
+  <th>Range</th>
+  <td><i>user-defined</i></td>
+</tr>
+<tr>
+  <th>Default</th>
+  <td><i>n/a</i></td>
+</tr>
```
Comment (uditgaurav) [inline docs/chaosengine-concepts.md]: By default, we use to give our image?

File: docs/chaosengine-concepts.md [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+  <th>Description</th>
+  <td>Flag to specify image of chaos runner pod</td>
+</tr>
+<tr>
+  <th>Type</th>
+  <td>Optional</td>
+</tr>
+<tr>
+  <th>Range</th>
+  <td><i>user-defined</i></td>
```
Comment (uditgaurav) [inline docs/chaosengine-concepts.md]: user-defined (type: string)

File: docs/chaosengine-concepts.md [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+  <th>Description</th>
+  <td>Flag to specify image pull policy for the chaos runner</td>
+</tr>
+<tr>
+  <th>Type</th>
+  <td>Optional</td>
+</tr>
+<tr>
+  <th>Range</th>
+<td><code>Always</code>, <code>IfNotPresent</code></td>
```
Comment (uditgaurav) [inline docs/chaosengine-concepts.md:313]: The Range of image pull policy can be never also. In such a case, we should add one line in the notes pointing it should not be never.

File: docs/chaosengine-concepts.md [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+  <th>Type</th>
+  <td>Mandatory</td>
+</tr>
+<tr>
+  <th>Range</th>
+  <td><code>active</code>, <code>stop</code></td>
+</tr>
+<tr>
+  <th>Default</th>
+  <td><code>active</code></td>
```
Comment (uditgaurav) [inline docs/chaosengine-concepts.md:37]: If we have given the default value i.e `active` then why it is mandatory?

Comment (uditgaurav) [reply docs/chaosengine-concepts.md:37]: unlike jobCleanupPolicy - which is optional

File: docs/chaosengine-concepts.md (hunk 3 above)
Comment (ksatchit) [reply docs/chaosengine-concepts.md]: this is no longer a part of the default example specs right now, as can be seen on chaos-charts. Also the default varies based on operator manifest, i.e., we enforce a default inside code and then on manifest. For the user of the chaosengine -- either of the above can be a "default". Added a note.

File: docs/chaosengine-concepts.md (hunk 2 above)
Comment (ksatchit) [reply docs/chaosengine-concepts.md:77]: Added this.

File: docs/chaosengine-concepts.md (hunk 1 above)
Comment (ksatchit) [reply docs/chaosengine-concepts.md:274]: Don't plan to expose this to users yet. As the monitoring infra/strategy needs to evolve better

File: docs/chaosengine-concepts.md (hunk 4 above)
Comment (ksatchit) [reply docs/chaosengine-concepts.md]: Done!

File: docs/chaosengine-concepts.md (hunk 5 above)
Comment (ksatchit) [reply docs/chaosengine-concepts.md:313]: We are exposing a chaosengine "range" of pull policies supported by Litmus/Chaosengine for the runner as only Always/IfNotPresent. Never will not supported at all us -- we should enforce this via CRD validation.

File: docs/chaosengine-concepts.md (hunk 6 above)
Comment (ksatchit) [reply docs/chaosengine-concepts.md:37]: Here mandatory refers to the *presence of the chaosengine flag* in the spec - not just the values. `engineState` will be mandatory to enforce actual state of engine.

**Review by uditgaurav — APPROVED** [review]
lgtm

**Review by ispeakc0de — APPROVED** [review]
/lgtm

**Review by ispeakc0de — APPROVED** [review]
/lgtm

**Review by chandankumar4 — APPROVED** [review]

**Pull request merged**